File: src/core/services/overlay_glass_settings.rs
pub struct OverlayGlassSettings;

impl OverlayGlassSettings {
    pub const DEFAULT_GLASS_CLARITY: i32 = 75;

    pub fn normalize_glass_clarity(clarity: i32) -> i32 {
        clarity.clamp(0, 150)
    }

    pub fn background_tint_opacity_from_clarity(clarity: i32) -> f64 {
        let normalized = Self::normalize_glass_clarity(clarity) as f64;
        // 0 -> 0.85 (solid), 150 -> 0.0 (clear)
        0.85 - (normalized / 150.0) * 0.85
    }

    pub fn blur_radius_from_clarity(clarity: i32) -> f64 {
        let normalized = Self::normalize_glass_clarity(clarity) as f64;
        // 0 -> 0 (solid), 150 -> 48 (max blur)
        (normalized / 150.0) * 48.0
    }

    pub fn content_opacity() -> f64 {
        1.0
    }
}

File: .برنامه ویندوزی/AWAYRA-WPF-main (2)/src/Awayra.Core/Services/OverlayGlassSettings.cs
namespace Awayra.Core.Services;

public static class OverlayGlassSettings
{
    public const int MinGlassClarity = 0;
    public const int MaxGlassClarity = 150;
    public const int DefaultGlassClarity = 100;
    public const double MinBlurRadius = 8.0;
    public const double MaxBlurRadius = 30.0;

    public static int NormalizeGlassClarity(int value) =>
        Math.Clamp(value, MinGlassClarity, MaxGlassClarity);

    public static int NormalizeGlassClarity(double value) =>
        NormalizeGlassClarity((int)Math.Round(value));

    public static double BackgroundTintOpacityFromClarity(int glassClarity)
    {
        var normalized = NormalizeGlassClarity(glassClarity);
        if (normalized <= 100)
        {
            return 1.0 - normalized / 100.0;
        }

        return 0.0;
    }

    public static double BlurRadiusFromClarity(int glassClarity)
    {
        var normalized = NormalizeGlassClarity(glassClarity);
        double radius;
        if (normalized <= 100)
        {
            radius = 30.0 - normalized * 0.12;
        }
        else
        {
            radius = 18.0 - ((normalized - 100.0) / 50.0) * 10.0;
        }

        return Math.Clamp(radius, MinBlurRadius, MaxBlurRadius);
    }

    public static int MigrateFromGlassTransparency(int glassTransparency) =>
        NormalizeGlassClarity(glassTransparency);

    public static int MigrateFromBackgroundVisibility(int backgroundVisibility)
    {
        if (backgroundVisibility <= 10)
        {
            return 0;
        }

        if (backgroundVisibility >= 30)
        {
            return 100;
        }

        if (backgroundVisibility <= 20)
        {
            return (int)Math.Round((backgroundVisibility - 10) * 5.0);
        }

        return (int)Math.Round(50 + (backgroundVisibility - 20) * 5.0);
    }

    public static int MigrateFromLegacyOpacity(double legacyOpacity)
    {
        if (double.IsNaN(legacyOpacity) || double.IsInfinity(legacyOpacity))
        {
            return DefaultGlassClarity;
        }

        var visibility = (int)Math.Round((1.0 - legacyOpacity) * 100.0);
        return MigrateFromBackgroundVisibility(Math.Clamp(visibility, 10, 30));
    }

    public static double ContentOpacity => 1.0;
}
